src/auth/auth.controller.ts

```diff
@@ -31,4 +31,41 @@ export class AuthController {
     async register(@Body() createUserDto: CreateUserDto): Promise<SanitizedUser> {
         return await this.authService.register(createUserDto);
     }
+
+    // POST /auth/confirm/:token
+    // This path lets the user confirm his email
+    @Public()
+    @Post('confirm')
+    async confirmEmail(@Query('token') token: string): Promise<string> {
```

Comment:
Using `@Param()` is preferable to `@Query()` when possible. Should be useful to enforce calling a route with a given parameter

```suggestion
    @Post('confirm/:token')
    async confirmEmail(@Param('token') token: string): Promise<string> {
```